lib/os/onoff.c

```diff
@@ -415,19 +417,14 @@ static void process_event(struct onoff_manager *mgr,
 	k_spin_unlock(&mgr->lock, key);
 }
 
-int onoff_request(struct onoff_manager *mgr,
-		  struct onoff_client *cli)
+static int request(struct onoff_manager *mgr,
```

Comment:
I'd like this to be named `request_and_unlock`, but at minimum this should document that the lock will be released when the function returns.